fuse/src/main/java/com/github/kittinunf/fuse/core/Cache.kt

```diff
@@ -64,13 +68,13 @@ class Cache<T : Any> internal constructor(private val config: Config<T>) : Fuse.
         memCache.get(safeKey)?.let { value ->
             // move specific key in disk cache up as it is found in mem
             val result = Result.of<T, Exception> {
-                val converted = convertToData(value as T)
                 if (diskCache.get(safeKey) == null) {
+                    val converted = convertToData(value as T)
```

Comment:
This line was moved into the `if` condition, because the `convertToData` operation is potentially expensive. Users of your library supply the converter, so "we" can't make assumptions about how fast it is.

It seems better to defer the data conversion until we _absolutely_ know we need to do it. Previously, the conversion would have run **even if `diskCache.get(safeKey)` returned a non-null value.**

Reply:
Ah, makes sense!